src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,46 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(animal) {
+    if (
+      animal instanceof Herbivore &&
+      !animal.hidden &&
+      !(animal instanceof Carnivore)
+    ) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        animal.die();
```

Comment:
[CODE PERFORMANCE] - It is preferred to filter animals with health less than or equal to 0 from `Animal.alive` array using Array's `filter` method. Instead of checking each time in the `bite` method, you could create a separate method to filter out dead animals and call it after the health is reduced.